src/core/ext/transport/chttp2/transport/internal.h

```diff
@@ -462,6 +463,8 @@ struct grpc_chttp2_transport : public grpc_core::KeepsGrpcInitialized {
   /// grace period to wait for data after sending a ping before keepalives
   /// timeout
   grpc_core::Duration keepalive_timeout;
+  /// number of stream objects currently allocated by this transport
+  std::atomic<size_t> streams_allocated{0};
```

Comment:
does this need to be atomic?

Reply:
init_stream (which uniquely calls the constructor for `grpc_chttp2_stream`) is called outside the combiner;

To keep this code simple I don't want to reason about which path we're taking to get to the constructor (since it's weirdly different between client & server, etc...) - doing so would only lead to bugs (we don't want a security problem because we accidentally decremented too many times).

Other places we use similar atomics (transport level):
- every transport op
- every time a stream is created or destroyed
- every time a stream is removed
- every ping
- every read
- every write

Also at the stream level:
- every time a stream becomes writable
- every stream op (at least 2x)

This is not including any of the refcounting that happens external to the transport.

The atomic is not a concern.

Reply:
Can you point me to these atomics? I don't spot them immediately.

I still think we shouldn't be using atomics unnecessarily, but given that the transport is going through overhaul, I'm ok with this for now.